cpp-examples/mutability.cpp

```diff
@@ -0,0 +1,31 @@
+#include <vector>
+#include <iostream>
+
+void print_shopping_list(std::vector<std::string>* shopping_list) {
+    if (shopping_list->size() == 0) {
+        std::cout << "\t* nothing\n";
+    }
+
+    for (auto item = shopping_list->begin(); item != shopping_list->end(); ++item) {
+        std::cout << "\t* " << *item << '\n';
+    }
+}
+
+void buy_groceries(std::vector<std::string> shopping_list) {
```

Comment:
is the copy intended?

Reply:
in this case, yeah, to keep it closer to https://github.com/hniemeyer/RustForCppDevsCodeSnippets/blob/main/rust-examples/src/mutability_solution.rs . I battled with this too, as it doesn't make much sense like this...